src/connection/connection.rs

```diff
@@ -6,26 +6,32 @@ use tokio::net::TcpStream;
 
 use crate::codec::{decode, encode, Token};
 use crate::command::{Command, CommandError, CommandProcessor};
+use crate::server::Context;
 
 pub type ConnectionId = u64;
 
 pub struct Connection {
     id: ConnectionId,
     socket: TcpStream,
     addr: SocketAddr,
+    context: Context,
 }
 
 impl Connection {
-    pub fn new(id: ConnectionId, socket: TcpStream, addr: SocketAddr) -> Self {
-        Connection { id, socket, addr }
+    pub fn new(ctx: Context, id: ConnectionId, socket: TcpStream, addr: SocketAddr) -> Self {
+        Connection {
+            id,
+            socket,
+            addr,
+            context: ctx,
```

Comment:
Ironically, trying to abbreviate `context` as `ctx` is making your code more verbose than it needs to be.

Reply:
Yep! I am in this habit from Go. I don't have a great reason to shorten it and I don't know of a Rust idiom for shortening it, so I've updated it to the full word in b93d1cb.